registry.go

```diff
@@ -5,13 +5,33 @@ import (
 	"sync"
 )
 
-var Tasks taskRegistry
+var Tasks TaskMap
 
-type taskRegistry struct {
+type TaskRegistry interface {
+	Get(name string) *Task
+}
+
+func MessageHandler(registry TaskRegistry) func(msg *Message) error {
```

Comment:
I think think should be a method on TaskRegistry, i.e. TaskRegistry should implement MessageHandler interface. Then instead of `taskq.MessageHandler(q.opt.Tasks)` you just write `q.opt.Tasks.MessageHandler`.

Reply:
This is linked to the interface argument above. If we decide extensibility via an interface is a good thing, then it's better to have a smaller interfaces. Since `HandleMessage` on `TaskRegistry` used to have an effectively static derived implementation once you have `Get(name string) *Task` then all an implementer of a `TaskGetter` (perhaps a better name, unless we could just pass the getter function itself in `QueueOptions`) has to implement is one methods and another one with a copy-and-paste implementation.

If you want to dispense with the interface then we may as well have the convenience function on `TaskRegistry` I suppose.